### ISSUE TITLE
Saving blocks does not work in SilverStripe 3.1.16

### ISSUE THREAD

**Issue by Taitava** [sheadawson/silverstripe-blocks#55]
Goto some page, create a new Block and try to save it for the first time (no matter if you are saving a draft or directly publishing). This occurs when saving:

```
[Error] Couldn't run query: INSERT INTO "Block_versions" ("ClassName", "Title", "CanViewType", "LastEdited", "Created", "Name", "Header", "Content", "Link", "VideoURL", "Template", "Active", "ExtraCSSClasses", "Weight", "Area", "Published", "Version", "RecordID", "AuthorID") VALUES ('SmallBlock', 'MyTitle', 'Anyone', '2016-02-08 21:52:54', '2016-02-08 21:52:54', null, 'None', null, null, null, null, '1', null, '0', null, '0', 1, 11, 1) Unknown column 'Name' in 'field list'
/*path to site root*/framework/model/MySQLDatabase.php:598 
```
'SmallBlock' is my custom block class. Suddenly a Name column appears from somewhere, but it does not exist in the database, and as far I know, it also should not exist. As you can see, the above query is assigning _null_ to that column anyway.

I'm very insecure about this issue as I don't know how to properly hunt it down and also don't have time to do it. I don't even know if this is a bug in this module at all, or if it's a bug in Versioned or in some other place in the framework.

In case it helps, here is the custom made SmallBlock class, but I don't think it relates to this bug:
```PHP
class SmallBlock extends Block
{
	private static $singular_name = 'Pieni laatikko';
	private static $plural_name = 'Pienet laatikot';

	private static $db = array(
		'Content'	=> 'HTMLText',
	);

	private static $has_one = array(
		'Image'		=> 'Image',
	);
}
```

After upgrading to SilverStripe 3.2.* the bug was gone. So if no other solution is found, I think it would be good enough to update composer.json and require SS 3.2.*.

**Comment by sheadawson** [comment]
I think the only explanation for this is that at some point you must have been using a slightly old version of the blocks module that still had the "Name" DB field. Otherwise SilverStripe would not be trying to write to that table column. Perhaps your blocks module was also updated when updating framework/cms composer dependencies? Or a dev/build was required? I can't see any reason why this error would occur on 3.1 so will close for now.

**Comment by Taitava** [comment]
Just for record to this old issue: Now when I returned to the project where I use this module, I spent a few hours scratching my head with this (as I don't like to solve it by upgrading SilverStripe _before_ I find out the real cause behind a bug in order to prevent it in the future).

All code was correct and my version of the blocks module was not outdated. Instead, the problem was in the database structure. I checked the structure and saw the old `Name` column in the `Block` table. I removed the column from the table, tried saving again a new `Block` in the CMS, and ran into the same error message again, but this time with a different column name. So I proceeded to remove columns one by one until the saving started to work.

These are the columns I removed: `Name`, `Header`, `Content`, `Link`, `VideoURL`, `Template` and `Active`

So my conclusion is that your module works flawlessly, but there is a bug in the `Versioned` class (and I guess they fixed it in SS 3.2). Any column that existed in the `Block` table were used in the `INSERT INTO Block_versions` query without first ensuring that the column actually exist in `Block_versions` too. As those columns did not exist, the saving process failed.

So anybody still using SS 3.1 can run into this problem by first installing an old version of this module and then upgrading the module to the current version. The upgraded module itself does nothing bad, it just abandons columns and implements a support for `Versioned`, which naturally will not create the obsolete columns into the `Block_versions` table. Problem solved :).

Thank you @sheadawson for pointing me to the right direction! :)
